st2reactor/tests/integration/test_garbage_collector.py

```diff
@@ -175,6 +191,50 @@ def test_garbage_collection(self):
         stderr_dbs = ActionExecutionOutput.query(output_type='stderr')
         self.assertEqual(len(stderr_dbs), (new_executions_count))
 
+    def test_inquiry_garbage_collection(self):
+        now = date_utils.get_datetime_utc_now()
+
+        # Insert some mock Inquiries with start_timestamp > TTL
+        old_inquiry_count = 15
+        timestamp = (now - datetime.timedelta(minutes=3))
+        for index in range(0, old_inquiry_count):
+            self._create_inquiry(ttl=2, timestamp=timestamp)
+
+        # Insert some mock Inquiries with start_timestamp < TTL
+        new_inquiry_count = 5
+        timestamp = (now - datetime.timedelta(minutes=3))
+        for index in range(0, new_inquiry_count):
+            self._create_inquiry(ttl=15, timestamp=timestamp)
+
+        filters = {
+            'status': action_constants.LIVEACTION_STATUS_PENDING
+        }
+        inquiries = list(ActionExecution.query(**filters))
+        self.assertEqual(len(inquiries),
+                         (old_inquiry_count + new_inquiry_count))
+
+        # Start garbage collector
+        process = self._start_garbage_collector()
+
+        # Give it some time to perform garbage collection and kill it
+        eventlet.sleep(10)
+        process.send_signal(signal.SIGKILL)
+        self.remove_process(process=process)
+
+        # Expired Inquiries should have been garbage collected
+        inquiries = list(ActionExecution.query(**filters))
+        self.assertEqual(len(inquiries), new_inquiry_count)
```

Comment:
Probably also good to assert on status - that it actually marks it as failed / timeout or whatever the status we want to use.

Reply:
I can do this to be more explicit, but I am already passing status into `filters`, so the `inquiries` list **should** only contain executions with the appropriate status.